### PR TITLE
overlord: keep tasks unlinked from a change hidden, prune them

### PR DESCRIPTION
**Pull request by pedronis**
With this tasks not linked to a Change are hidden from State.Task and State.Tasks (used by taskrunner), and they are also pruned. Usually they are the result of the setting up of a Change that failed early.

### CODE REVIEW

File: overlord/state/state.go [hunk 1]
```diff
@@ -341,6 +355,13 @@ func (s *State) Prune(pruneWait, abortWait time.Duration) {
 			delete(s.changes, chg.ID())
 		}
 	}
+	for tid, t := range s.tasks {
+		// TODO: this could be done more aggressively
+		if t.Change() == nil && t.SpawnTime().Before(pruneLimit) {
+			s.writing()
+			delete(s.tasks, tid)
```
Comment (chipaca) [inline overlord/state/state.go]: would it make sense to log when this happens?

**Comment by chipaca** [comment]
LGTM, with a Q.

File: overlord/state/state.go (hunk 1 above)
Comment (pedronis) [inline overlord/state/state.go]: @chipaca it's a natural occurrence of any path that creates some tasks then discovers it needs to error out, it's ok to try to postpone making changes, but posting making tasks themselve acrosss multilple handlers is a pain, we could log it at debug level

**Comment by mvo5** [comment]
:+1:

**Pull request merged**